### PR TITLE
Feature/appsecret proof

### PR DESCRIPTION
**Pull request by cleder**
This is the continuation from #409 to address the required code review changes

### CODE REVIEW

File: facebook/__init__.py [hunk 1]
```diff
@@ -94,6 +95,14 @@ def __init__(
         self.timeout = timeout
         self.proxies = proxies
         self.session = session or requests.Session()
+        self.app_secret_hmac = None
+
+        if app_secret and self.access_token:
+            # Generates an app secret hmac based on
+            # https://developers.facebook.com/docs/graph-api/securing-requests
+            self.app_secret_hmac = create_appsecret_proof(
```
Comment (martey) [inline facebook/__init__.py]: Since the `create_appsecret_proof` method is only called from this one location, does it need to be a separate method?

**Comment by thieman** [comment]
It seems like 5+ people have been working on some form of this PR since 2016. Does the value of merging it now not outweigh whatever minor concerns remain?

**Comment by DmytroLitvinov** [comment]
Hello @martey 

With new GraphAPI version, appsecret proof will be required for some of API.
If you need help with refactoring with MR, please let me know and I will prepare new PR.

**Comment by martey** [comment]
[I provided feedback 6 months ago](https://github.com/mobolic/facebook-sdk/pull/445#pullrequestreview-193833309) about the one remaining issue that needed to be resolved before this can be merged. At this point, I consider this pull request abandoned, so @DmytroLitvinov (or anyone else wanting this merged), you do not need my permission to work on this.

**Comment by martey** [comment]
See #452 and #458.

**Pull request closed without merging**